app/marketing/management/commands/assemble_leaderboards.py

```diff
@@ -36,6 +36,50 @@
 yearly_cutoff = timezone.now() - timezone.timedelta(days=365)
 
 
+def profile_to_location(handle):
+    profiles = Profile.objects.filter(handle__iexact=handle)
+    if handle and profiles.exists():
+        profile = profiles.first()
+        return profile.locations
+    return []
+
+
+def bounty_to_location(bounty):
+    locations = profile_to_location(bounty.bounty_owner_github_username)
+    fulfiller_usernames = list(bounty.fulfillments.filter(accepted=True).values_list('fulfiller_github_username', flat=True))
+    for username in fulfiller_usernames:
+        locations = locations + locations
+    return locations
+
+
+def tip_to_location(tip):
+    return profile_to_location(tip.username) + profile_to_location(tip.from_username)
+
+
+def tip_to_country(tip):
+    return list(set([ele['country_name'] for ele in tip_to_location(tip) if ele and ele['country_name']]))
```

Comment:
Maybe we could adjust this to query the profile and fetch locations once and iterate over these checks against the already retrieved data? We could potentially define tip/bounty to_something once and pass the ele key to check for as well. Either way, this should work.

Reply:
i strongly agree... it seems like the ip to location stuff is pretty unperformant

Reply:
@mbeacom how do you see this working?  should we just have a `Location` model that we `fk` to the Profile object when there is a new IP / location introduced, and just treat the DB as the source of truth for this?

Reply:
Hhmm - we could do that.  Will we want to track the location on every interaction?  We could add a location relation onto `UserAction` and/or drop a location fk on profile.